### PR TITLE
Add Firebase Segmentation SDK structure.

### PR DESCRIPTION
**Pull request by dmandar**

### CODE REVIEW

File: FirebaseSegmentation.podspec [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#
+# Be sure to run `pod lib lint FirebaseSegmentation.podspec' to ensure this is a
+# valid spec before submitting.
+#
+# Any lines starting with a # are optional, but their use is encouraged
+# To learn more about a Podspec see https://guides.cocoapods.org/syntax/podspec.html
+#
```
Comment (morganchen12) [inline FirebaseSegmentation.podspec]: nit: These comments should be removed

File: FirebaseSegmentation.podspec [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Pod::Spec.new do |s|
+  s.name             = 'FirebaseSegmentation'
+  s.version          = '0.1.0'
+  s.summary          = 'Firebase Segmentation SDK for user segmentation.'
```
Comment (morganchen12) [inline FirebaseSegmentation.podspec]: nit: `for user segmentation` is redundant and can be removed

File: FirebaseSegmentation.podspec [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+# This description is used to generate tags and improve search results.
+#   * Think: What does it do? Why did you write it? What is the focus?
+#   * Try to keep it short, snappy and to the point.
+#   * Write the description between the DESC delimiters below.
+#   * Finally, don't worry about the indent, CocoaPods strips it!
```
Comment (morganchen12) [inline FirebaseSegmentation.podspec]: nit: Remove these comments

File: Segmentation/Public/FIRSegmentation.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ *  @return A shared instance of FIRSegmentation.
+ */
++ (nonnull FIRSegmentation *)segmentation NS_SWIFT_NAME(segmentation());
+
+/// Singleton instance (scoped to FIRApp)
+/// Returns the FIRSegmentation instance for your Firebase application. This singleton class
+/// instance lets you set your own custom identifier to be used for targeting purposes within
+/// Firebase.
+//@@+ (nonnull FIRSegmentation *)segmentationWithApp:(nonnull FIRApp *)app
+// NS_SWIFT_NAME(segmentation(app:));
```
Comment (morganchen12) [inline Segmentation/Public/FIRSegmentation.h]: Is there a reason this is commented out?

File: Segmentation/Public/FIRSegmentation.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ */
+- (instancetype)init __attribute__((unavailable("Use +segmentation instead.")));
+
+/// Set your own custom installation ID to be used for segmentation purposes.
+/// This method needs to be called every time (and immediately) upon any changes to the custom
+/// installation ID.
+/// @param completionHandler Set custom installation ID completion.
+/// @param setCustomInstallationIDError nil if initialization succeeded.
+- (void)setCustomInstallationId:(NSString *)customInstallationId
+                 withCompletion:(void (^)(NSError *setCustomInstallationIDError))completionHandler;
```
Comment (morganchen12) [inline Segmentation/Public/FIRSegmentation.h]: nit: Name this `completion` instead of `withCompletion`

File: FirebaseSegmentation.podspec [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    :tag => 'Segmentation-' + s.version.to_s
+  }
+
+  s.ios.deployment_target = '8.0'
+  
+  s.cocoapods_version = '>= 1.4.0'
+  s.static_framework = true
+  s.prefix_header_file = false
+
+  s.source_files = 'Segmentation/**/*'
```
Comment (paulb777) [inline FirebaseSegmentation.podspec]: Test files should not be included in the library sources.

**Pull request closed without merging**